lib/features/dind.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * This module handles the creations of a dind-service container, links and sets
+ * the DOCKER_HOST environment variables such that task container will be able
+ * to use `docker`. For security reasons you can't run privileged containers,
+ * use capabilities or volume mount host folders. But it should work out of the
+ * box for common docker flows.
+ */
+
+import _ from 'lodash';
+import waitForSocket from '../wait_for_socket';
+import { pullImageStreamTo } from '../pull_image_to_stream';
+import slugid from 'slugid';
+import rmrf from 'rimraf';
+import path from 'path';
+import fs from 'fs';
+let debug = require('debug')('docker-worker:features:dind');
+
+// Maximum time to wait for dind-service to be ready
+const INIT_TIMEOUT = 30 * 1000;
+
+export default class DockerInDocker {
+  constructor() {
+    // dind-service container
+    this.container = null;
+    this.tmpFolder = path.join('/tmp', slugid.v4());
+  }
+
+  async link(task) {
```

Comment:
oh, yeah, I considered it when I started... Now I can't decide...
Technically it's locked down, so there not really any reason to scope it.

However, I suspect we'll add scopes when we make dind-cache folders... But then we might want to allow things run without persistent cache folders.

Only reason people shouldn't use this all the time is that pulling down docker images is not reliable...